backend/infrahub/graphql/mutations/proposed_change.py

```diff
@@ -95,6 +95,19 @@ async def mutate_update(
     ):
         context: GraphqlContext = info.context
 
+        has_merge_permission = False
+        if context.account_session:
```

Comment:
I added a property `context.active_account_session` that raises an error if `context.account is None`. At some point I'd like to remove the account_session as optional on the GraphqlContext object. Even with an anonymous user we should have that context set. I think if you change this to a context.active_account_session we can get rid of this if/else statement and potentially remove that "disable=too-many-branches"

Reply:
Agreed, [I did that while I was experimenting](https://github.com/opsmill/infrahub/pull/4443/commits/19f7e783bf02d36ef0332a81602da758d5a470aa). However since it's going to raise an exception when `account_session` is not set, it's going to make a bunch of tests fail. They trigger the mutation thanks to the `graphql_mutation` helper and do not necessarily pass an account session to it. So we'd have to adjust some of the tests to avoid this, but I think it should be done in another PR.